modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -13,15 +13,13 @@ def do_create(name, value, type)
       zone = match_zone(name, enum_zones)
       msg = "Added #{type} entry #{name} => #{value}"
       value = "#{value}." if type == "PTR"
-      cmd = "/RecordAdd #{zone} #{name}. #{type} #{value}"
-      execute(cmd, msg)
+      execute("/RecordAdd", zone, "#{name}.", type, value, msg: msg, error_only: false)
       nil
     end
 
     def remove_specific_record_from_zone(zone_name, node_name, record, type)
       msg = "Removed #{record} #{type} record #{node_name} from #{zone_name}"
-      cmd = "/RecordDelete #{zone_name} #{node_name}. #{type} #{record} /f"
-      execute(cmd, msg)
+      execute("/RecordDelete", zone_name, "#{node_name}.", type, "#{record} /f", msg: msg, error_only: false)
```

Comment:
One more thing I noticed, every argument must be a separate array element. Here the `/f` argument is passed with a record string. There is difference if you call an executable with:

   executable RECORD /f

or

    executable "RECORD /f"

The first one is two arguments (that is expected) however the second form is a single argument with "RECORD /f" string, that would fail actually.

Reply:
I see. I misunderstood the meaning of that parameter. I am about to fix it.